flang/lib/Optimizer/Dialect/FIRType.cpp

```diff
@@ -178,8 +178,9 @@ SequenceType parseSequence(mlir::DialectAsmParser &parser, mlir::Location) {
   return SequenceType::get(shape, eleTy, map);
 }
 
-static bool verifyIntegerType(mlir::Type ty) {
-  return ty.isa<mlir::IntegerType>() || ty.isa<IntType>();
+/// Is `ty` a standard or FIR integer type?
+static bool isaIntegerType(mlir::Type ty) {
```

Comment:
Could `isa_integer` from FIRType.h be used here instead of defining this local helper (I know it is different given `isa_integer` accepts `mlir::IndexType` and `isaIntegerType` does not, but maybe it is actually better if `isaIntegerType` also accepts those.)

Reply:
Yep. I'll add a TODO to investigate it further.